fat/fat2/transactionbatch.go

```diff
@@ -0,0 +1,141 @@
+package fat2
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Factom-Asset-Tokens/factom"
+	"github.com/Factom-Asset-Tokens/fatd/fat"
+)
+
+// Transaction represents a fat2 transaction, which can be a list of one or more
+// sub-transactions to be executed in order
+type TransactionBatch struct {
+	Version      uint          `json:"version"`
+	Transactions []Transaction `json:"transactions"`
+	fat.Entry
+}
+
+// NewTransactionBatch returns a TransactionBatch initialized with the given entry.
+func NewTransactionBatch(entry factom.Entry) TransactionBatch {
+	return TransactionBatch{Entry: fat.Entry{Entry: entry}}
+}
+
+type transactionBatch TransactionBatch
+
+// MarshalJSON marshals the TransactionBatch content field as JSON, but will raise an error if
+// the batch fails the checks in ValidData()
+func (t TransactionBatch) MarshalJSON() ([]byte, error) {
+	if err := t.ValidData(); err != nil {
```

Comment:
I do this but Idk if I want to anymore. See my messages in discord. we'll leave it for now for consistency until I decide to change it all.

Reply:
Alright, I'll follow up on discord with your messages. Resolving this